### ISSUE TITLE
Create S3 buckets in AWS regions

### ISSUE THREAD

**Issue by jaypipes** [kubernetes/k8s.io#3595]
- [x] Naming convention for buckets will be `${prefix}registry-k8s-io-${gcs_bucket_name}-${region}`
- [x] Create buckets in each AWS region listed [here](https://github.com/kubernetes-sigs/oci-proxy/issues/38#issuecomment-1085369923) via Terraform
- [x] Ted Z to vet the IAM role that will have permission to create buckets and write objects

**Comment by hh** [comment]
Maybe put the code within `k/k8s.io infra/aws/registry.k8s.io/buckets` for now.

**Comment by Riaankl** [comment]
Form the [data](https://github.com/kubernetes-sigs/oci-proxy/issues/39#issuecomment-1085230292) these are the regions:

us-west-2
us-west-1
us-east-1
eu-central-1
eu-central-1
us-east-2
ap-southeast-1
us-west-1
ap-northeast-1
ap-south-1

**Comment by Riaankl** [comment]
The PR address the teraform of S3 buckets.
#3605

**Comment by hh** [comment]
@BobyMCbobs we need a project to run this in. To create the buckets
Breakdown

**Comment by hh** [comment]
@jaypipes working with Adolfo to run the mirroring.
Jay has what he needs. Unblocked currently.

@arnaud >
 
We should do a one sync copy from GCS to S3... 10 buckets is going to take a while.

If we do the one sync, then later use the mirror to push these options. It's going to be easier, as we already have a copy of the source of truth over s3 so we don't waste time.
@dims >

We already have one is a single s3 bucket, we just need to copy to one.
We need to create one bucket and copy to the s3 bucket.

@arnaud > There is one permission to be added to do the once off sync

> IAM role that image promoter will be using when doing the mirroring between GCS and S3

When we have that first bucket, then we can try some additional stuff that @BenTheElder is writing.

So we can redirect everything to that one bucket.... some prow jobs running inside AWS so we can test that.

- [ ] IAMs  roles for pushing is correct
- [ ] IAM permissions for pushing to buckets to correct
- [ ] The IAM role that's pushing that IAM needs IAM creation permissions
- [ ] Run terrafrom!

Not technically a blocker... between Caleb and Arnaud.

**Comment by hh** [comment]
Arnaud is going to create another issue, lets be sure it's on the board.

**Comment by ameukam** [comment]
> Arnaud is going to create another issue, lets be sure it's on the board.

@hh From https://kubernetes.slack.com/archives/CCK68P2Q2/p1650407982934869, I'm now not sure if we need to created a new issue.

**Comment by BobyMCbobs** [comment]
# Update for things
@jaypipes and I have been paring on setting up the accounts in a organized manner and are setting up a new account for the registry buckets to go into.

The structure is root/Kubernetes _(OU)_/registry.k8s.io _(OU)_/k8s-infra-aws-registry-k8s-io-admin@kubernetes.io _(account)_.

Currently in the middle of sorting IAM access to this account for provisioning the buckets and IAM role for accessing it.

Ticket for updating docs which reflect the org/account structuring: https://github.com/kubernetes/k8s.io/pull/3668

**Comment by ameukam** [comment]
From meeting April 27,
Excepting production buckets by end of the week. If not create S3 buckets in the sandbox account.

**Comment by BobyMCbobs** [comment]
# Bucket update

Tried to do an initial password (re)set for the k8s-infra-aws-registry-k8s-io-admin@kubernetes.io AWS account.
@dims has been monitoring the emails from that address and no email has come through.

Will be paring with @jaypipes when next available to complete it.

**Comment by Riaankl** [comment]
Buckets created in https://github.com/kubernetes/k8s.io/pull/3693

**Comment by Riaankl** [comment]
add in eu-west-1 AWS region https://github.com/kubernetes/k8s.io/pull/3699

**Comment by ameukam** [comment]
@BobyMCbobs @jaypipes Can we get the full list of the buckets names and the ARNs of those buckets ?

**Comment by BobyMCbobs** [comment]
> @BobyMCbobs @jaypipes Can we get the full list of the buckets names and the ARNs of those buckets ?

@ameukam, the bucket names and ARNs are as follows
- arn:aws:s3:us-west-1:768319786644:prod-registry-k8s-io-us-west-1
- arn:aws:s3:us-west-2:768319786644:prod-registry-k8s-io-us-west-2
- arn:aws:s3:us-east-1:768319786644:prod-registry-k8s-io-us-east-1
- arn:aws:s3:us-east-2:768319786644:prod-registry-k8s-io-us-east-2
- arn:aws:s3:eu-central-1:768319786644:prod-registry-k8s-io-eu-central-1
- arn:aws:s3:ap-southeast-1:768319786644:prod-registry-k8s-io-ap-southeast-1
- arn:aws:s3:ap-northeast-1:768319786644:prod-registry-k8s-io-ap-northeast-1
- arn:aws:s3:ap-south-1:768319786644:prod-registry-k8s-io-ap-south-1

(this is me figuring out the ARNs so let me know if it works OK)

cc @Riaankl

**Comment by ameukam** [comment]
@BobyMCbobs Thanks!!! I took the liberty edit the AWS accounts to avoid potential attacks.

**Comment by jaypipes** [comment]
> @BobyMCbobs Thanks!!! I took the liberty edit the AWS accounts to avoid potential attacks.

I don't think there's anything particularly sensitive about the AWS account ID, @ameukam :)

**Comment by ameukam** [comment]
@BobyMCbobs Now the buckets are up, what are the blockers to sync from the blobs from k8s.gcr.io to all those buckets ?

**Comment by thockin** [comment]
Hey all, If this is done - who has access to these buckets and how do we govern that?  We really really need to move this forward ASAP.

What's blocking?

**Comment by BenTheElder** [comment]
xref: https://github.com/kubernetes/k8s.io/issues/3807 for a possible option to allow CI jobs to access AWS without storing creds once bootstrapped via workload identity.

Alternatively, if someone who has access currently can at least do a one-off sync https://github.com/kubernetes/k8s.io/issues/3623 of the current contents, we can at least make progress on rolling out "traffic to AWS" in registry.k8s.io while we finish figuring out the ongoing automated sync.

@ameukam can provide pointers on rclone from GCR's GCS to s3.

**Comment by Riaankl** [comment]
> Hey all, If this is done - who has access to these buckets and how do we govern that? We really really need to move this forward ASAP.
> 
> What's blocking?

/cc @jaypipes

**Comment by BobyMCbobs** [comment]
Depends on https://github.com/cncf-infra/aws-infra/issues/4

**Comment by ameukam** [comment]
/sig k8s-infra
/area infra
/area release-eng
/priority critical-urgent
/milestone v1.25
/close 

Follow-up:
  - https://github.com/cncf-infra/aws-infra/issues/4
  - https://github.com/kubernetes/k8s.io/issues/3623

**Comment by thockin** [comment]
Why close this if it is not accessible yet?

**Comment by ameukam** [comment]
> * registry-k8s-io IAM role set up cncf-infra/aws-infra#4

The purpose of the issue was to make sure the buckets exist whether private or public. anonymous access to those is a different goal and I think we should direct future questions to https://github.com/cncf-infra/aws-infra/issues/4 (currently admins access to those buckets is handled by CNCF).
